lib/game/game_widget.dart

```diff
@@ -1,100 +1,270 @@
-import 'package:flutter/gestures.dart';
 import 'package:flutter/widgets.dart';
+import 'package:flutter/gestures.dart';
 
-import '../components/mixins/tapable.dart';
-import '../gestures.dart';
-import 'embedded_game_widget.dart';
 import 'game.dart';
+import 'embedded_game_widget.dart';
+import '../gestures.dart';
+import '../components/mixins/tapable.dart';
 
-bool _hasBasicGestureDetectors(Game game) =>
-    game is TapDetector ||
-    game is SecondaryTapDetector ||
-    game is DoubleTapDetector ||
-    game is LongPressDetector ||
-    game is VerticalDragDetector ||
-    game is HorizontalDragDetector ||
-    game is ForcePressDetector ||
-    game is PanDetector ||
-    game is ScaleDetector;
+typedef GameLoadingWidgetBuilder = Widget Function(
+  BuildContext,
+  bool error,
+);
+
+/// A [StatefulWidget] that is in charge of attaching a [Game] instance into the flutter tree
+///
+class GameWidget<G extends Game> extends StatefulWidget {
```

Comment:
Should be T instead of G

Reply:
Why?

Reply:
It is the standard to denote type with `T`, I haven't seen that it would be any differently in Dart yet at least.
They use `T` here too: https://dart.dev/guides/language/effective-dart/style
It helps to know that the type is derived for generics and that it isn't the name of an existing type when reading the code.

Reply:
> use T, S, and U for generics that have a single type parameter and where the surrounding type makes its meaning obvious

`GameWidget` makes it pretty easy to understand that it is about a `Game`. Wilco thanx!